subprojects/signing/src/integTest/groovy/org/gradle/plugins/signing/SigningPublicationsIntegrationSpec.groovy

```diff
@@ -347,19 +345,56 @@ class SigningPublicationsIntegrationSpec extends SigningIntegrationSpec {
 
         then:
         executedAndNotSkipped(":publishIvyJavaPublicationToIvyRepository")
+        filesExists(this).forEach { it.assertExists() }
+        filesDoesNotExist(this).forEach { it.assertDoesNotExist() }
 
-        and:
-        ivyRepoFile(jarFileName).assertExists()
-        ivyRepoFile("${jarFileName}.asc").assertExists()
-        ivyRepoFile("ivy-${version}.xml").assertExists()
-        ivyRepoFile("ivy-${version}.xml.asc").assertExists()
-        ivyRepoFile("$artifactId-${version}-source.jar").assertExists()
-        ivyRepoFile("$artifactId-${version}-source.jar.asc").assertExists()
-        ivyRepoFile("$artifactId-${version}.module").assertDoesNotExist()
-        ivyRepoFile("$artifactId-${version}.module.asc").assertDoesNotExist()
 
-        and:
-        outputContains "Publication of Gradle Module Metadata is disabled because you have configured an Ivy repository with a non-standard layout"
+        where:
+        layout     | declaration | filesExists                                       | filesDoesNotExist
```

Comment:
💄 I'd rename the columns `filesExists => expectedFiles` and `filesDoesNotExist => unexpectedFiles`.